src/main/java/model/Database.java

```diff
@@ -530,18 +605,90 @@ public void updateVehicle(Vehicle v) throws Exception {
+            ps.close();
+            return;
+        }
+
+        //commit changes and close prepared statement
+        ps.close();
+
+        Log.log("Vehicle with vlicense " + v.vlicense + " successfully deleted");
     }
 
-    public List<Vehicle> getVehicleWith(VehicleType vt, Location l, boolean availableNow) throws Exception {
-        // TODO: implement this
-        throw new Exception("Method not implemented");
+    public List<Vehicle> getVehicleWith(VehicleType vt, Location l, Vehicle.VehicleStatus availableNow) throws Exception {
+        String query = Queries.Vehicle.GET_VEHICLES_WITH;
+
+        if (vt == null && l == null && availableNow == null){
+            //If no filters are provided, return all the results
+            query = query.replace(" WHERE vtName = ? AND location = ? AND city = ? AND status = ?", "");
+        } else {
+            if (vt == null) query = query.replace("vtName = ? AND ", "");
+            else query = query.replace("vtName = ?", "vtName = '" + vt.vtname + "'");
+            if (l == null) query = query.replace("location = ? AND city = ? AND ", "");
+            else query = query.replace("location = ? AND city = ?", "location = '" + l.location + "' AND city = '" + l.city + "'");
+            if (availableNow == null) query = query.replace("AND status = ?", "");
+            else query = query.replace("status = ?", "status = " + (availableNow == Vehicle.VehicleStatus.AVAILABLE));
+        }
+
+        System.out.println(query);
+        PreparedStatement ps = conn.prepareStatement(query);
+        ResultSet rs = ps.executeQuery();
+
+        List<Vehicle> vehicles = new ArrayList<Vehicle>();
+
+        while (rs.next()){
+            //Make a reservation object corresponding to a tuple queried from the database
+            Vehicle v = new Vehicle(rs.getInt("vId"),
+                                    rs.getString("vLicense"),
+                                    rs.getString("make"),
+                                    rs.getString("model"),
+                                    rs.getInt("year"),
+                                    rs.getString("color"),
+                                    rs.getInt("odometer"),
+                                    rs.getString("vtName"),
+                                    (rs.getBoolean("status")) ? Vehicle.VehicleStatus.AVAILABLE: Vehicle.VehicleStatus.RENTED,
+                                    new Location(rs.getString("city"), rs.getString("location")));
+
+            //Add the reservation object to r
+            vehicles.add(v);
+        }
+        ps.close();
+        return vehicles;
+
     }
 
     public Vehicle getVehicleMatching(Vehicle v) throws Exception {
-        // TODO: implement this
-        throw new Exception("Method not implemented");
+        PreparedStatement ps = conn.prepareStatement("SELECT * FROM Vehicle WHERE vLicense = '" + v.vlicense + "'");
```

Comment:
We should add the Query to the Query class and reference it here, rather than have queries all over the project